src/wanderer/lib.ts

```diff
@@ -0,0 +1,248 @@
+import { buy, canAdventure, Item, Location, use } from "kolmafia";
+import { $effect, $item, $location, $locations, clamp, get, have, sum } from "libram";
+import { NumericProperty } from "libram/dist/propertyTypes";
+import { realmAvailable } from "../lib";
+import { digitizedMonstersRemaining, estimatedTurns } from "../turns";
+
+export type DraggableFight = "backup" | "wanderer" | "yellow ray";
+
+interface UnlockableZone {
+  zone: string;
+  available: () => boolean;
+  unlocker: Item;
+  noInv: boolean;
+}
+
+export const UnlockableZones: UnlockableZone[] = [
+  {
+    zone: "Spaaace",
+    available: () => have($effect`Transpondent`),
+    unlocker: $item`transporter transponder`,
+    noInv: false,
+  },
+  {
+    zone: "Wormwood",
+    available: () => have($effect`Absinthe-Minded`),
+    unlocker: $item`tiny bottle of absinthe`,
+    noInv: false,
+  },
+  {
+    zone: "Rabbit Hole",
+    available: () => have($effect`Down the Rabbit Hole`),
+    unlocker: $item`"DRINK ME" potion`,
+    noInv: false,
+  },
+  {
+    zone: "Conspiracy Island",
+    available: () => realmAvailable("spooky"),
+    unlocker: $item`one-day ticket to Conspiracy Island`,
+    noInv: true,
+  },
+  {
+    zone: "Dinseylandfill",
+    available: () => realmAvailable("stench"),
+    unlocker: $item`one-day ticket to Dinseylandfill`,
+    noInv: true,
+  },
+  {
+    zone: "The Glaciest",
+    available: () => realmAvailable("cold"),
+    unlocker: $item`one-day ticket to The Glaciest`,
+    noInv: true,
+  },
+  {
+    zone: "Spring Break Beach",
+    available: () => realmAvailable("sleaze"),
+    unlocker: $item`one-day ticket to Spring Break Beach`,
+    noInv: true,
+  },
+];
+
+export function underwater(location: Location): boolean {
+  return location.environment === "underwater";
+}
+
+export function canAdventureOrUnlock(loc: Location): boolean {
+  const skiplist = [
+    ...$locations`The Oasis, The Bubblin' Caldera, Barrrney's Barrr, The F'c'le, The Poop Deck, Belowdecks, 8-Bit Realm, Madness Bakery, The Secret Government Laboratory, The Dire Warren`,
+    ...Location.all().filter((l) => l.parent === "Clan Basement"),
+  ];
```

Comment:
Is there any meaningful performance value in defining this (and the other skipLists) outside of the functions, or is that frivolous of me?